### ISSUE TITLE
[a11y]: Using dropdown with MacOS Voiceover and Firefox is impossible

### ISSUE THREAD

**Issue by tamaracha** [carbon-design-system/carbon#12995]
### Package

carbon-components

### Browser

Firefox

### Operating System

MacOS

### Package version

v11

### Automated testing tool and ruleset

no automated testing, encountered on a real website

### Assistive technology

Voiceover screenreader

### Description

When I try to open the focused dropdown by pressing arrow down, no options are announced by voiceover. If another input element is available, focus jumps to that input. Navigating with the voiceover cursor immediately closes the dropdown.

### Reproduction/example

Just open your [live demo](https://carbondesignsystem.com/components/dropdown/usage#live-demo), activate voiceover and try to use the dropdown.

### Steps to reproduce

Business websites are using this component which made it impossible for me to fill a form as a blind user.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
> Just open your [live demo](https://carbondesignsystem.com/components/dropdown/usage#live-demo), activate voiceover and try to use the dropdown.

@tamaracha I'm not sure if this is limited to the live demo on the website or a true bug within the component. Can you reproduce the issue on [the storybook iframe link of the dropdown story](https://react.carbondesignsystem.com/iframe.html?args=&id=components-dropdown--default&viewMode=story)?

**Comment by tamaracha** [comment]
Yes, the behaviour is similar.

- In safari, I can select via keyboard navigation, and I can partially use voiceover navigation. Here it could be seen as usable.
- In firefox, the options are not announced with keyboard navigation (up/down), and the dropdown closes when I try to explore the open dropdown via voiceover navigation.

Voiceover can be activated on Mac with cmd+F5

**Comment by guidari** [comment]
Based on the research I did, the behavior is related to the focus that VoiceOver applies when navigating the `ListBoxMenuItem` component.

The only solution I found to avoid this issue is by disabling a specific option in the VoiceOver settings:

1. Open the VoiceOver Utility in the system settings.
2. Click on 'Navigation' tab.
3. Disable the option called 'Synchronize keyboard focus and VoiceOver cursor'.

After applying this setting, when you test the application, you will notice that the focus is no longer applied to the `ListBoxMenuItem`, and VoiceOver correctly reads the elements inside it.

I attempted a few solutions using CSS and ARIA attributes, but unfortunately, none of them worked in this case.
